### ISSUE TITLE
Error when creating the conda envirnment from the environment.yml file.

### ISSUE THREAD

**Issue by ChristosPeridis** [tmdt-buw/karolos#5]
Hello dear Christian and dear Thunfisch,

I hope you and everyone in your family are all doing well! I am a researcher in Deep Reinforcement Learning and I was very interested to use the Karolos API for conducting Reinforcement Learning experiments in robotics. I followed the instructions on the main Git-Hub page in order to create the appropriate conda environment from the 'environment.yml'. However, when running the conda command for creating the environment from the 'environment.yml' file I received the following error message:

Installing pip dependencies: / Ran pip subprocess with arguments:            
['/home/cocp5/anaconda3/envs/karolos/bin/python', '-m', 'pip', 'install', '-U', '-r', '/home/cocp5/karolos/condaenv.ivei66uj.requirements.txt', '--exists-action=b']                                                                   
Pip subprocess output:                                                       
Collecting gym==0.18.0                                                       
  Downloading gym-0.18.0.tar.gz (1.6 MB)                                     
                                                                             
Pip subprocess error:                                                        
    ERROR: Command errored out with exit status 1:                           
     command: /home/cocp5/anaconda3/envs/karolos/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-t9mm8biz/gym_3e3125c05389450aba633da1817bb5b2/setup.py'"'"'; __file__='"'"'/tmp/pip-install-t9mm8biz/gym_3e3125c05389450aba633da1817bb5b2/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-r0s48li1
         cwd: /tmp/pip-install-t9mm8biz/gym_3e3125c05389450aba633da1817bb5b2/
    Complete output (1 lines):
    error in gym setup command: 'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers.
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.

failed

CondaEnvException: Pip failed

How can I overcome this issue ?

I am always ate your disposal for any further information regarding my system's setup.

Thank you very much for your valuable help in advance !!!

Kind regards,

Christos Peridis

**Comment by chrisbitter** [comment]
Hello Christos,

Thank you for taking interest in our framework!

Not 100% sure what the cause of your problem is, but we'll figure it out.

Can you try another gym version, e.g. by deleting the version specification
(the '==0.18.0' in the environment.yml)?

Cheers, Christian

**Comment by chrisbitter** [comment]
I just tested with the newest gym version (0.26.2) and it worked for me. We should eventually switch to gymnasium, since gym is no longer maintained. I'll open a separate issue for this.

Closing this issue. Feel free to reopen if the error persists.

**Comment by ChristosPeridis** [comment]
Hello dear Christian,

Thank you very much for your immediate response! Today I cloned the repository again I managed to successfully create the conda environment with no issues. I also executed the "train_sac_panda_reach.py' file successfully and it is currently running, I am plotting the performance using tensor-board.

Many thanks again for your valuable help and support!!!

Kind regards,

Christos Peridis
